plugin/src/main/java/org/opensearch/ml/rest/RestMLUpdateModelAction.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.rest;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.ml.plugin.MachineLearningPlugin.ML_BASE_URI;
+import static org.opensearch.ml.utils.RestActionUtils.PARAMETER_MODEL_ID;
+import static org.opensearch.ml.utils.RestActionUtils.getParameterId;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.ml.common.transport.model.MLUpdateModelAction;
+import org.opensearch.ml.common.transport.model.MLUpdateModelInput;
+import org.opensearch.ml.common.transport.model.MLUpdateModelRequest;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+
+import com.google.common.collect.ImmutableList;
+
+public class RestMLUpdateModelAction extends BaseRestHandler {
+
+    private static final String ML_UPDATE_MODEL_ACTION = "ml_update_model_action";
+
+    @Override
+    public String getName() {
+        return ML_UPDATE_MODEL_ACTION;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return ImmutableList
+            .of(new Route(RestRequest.Method.POST, String.format(Locale.ROOT, "%s/models/_update/{%s}", ML_BASE_URI, PARAMETER_MODEL_ID)));
```

Comment:
Its better to use "PUT" method and avoid using _update in the path. You can refer to model_group update API. This convention has been followed by other plugins like AD